src/access.rs

```diff
@@ -0,0 +1,339 @@
+				*locked_queued_outputs = Vec::new();
+			}
+
+			{
+				// Query the interface for relevant txids and check whether they have been
+				// reorged-out of the chain.
+				let unconfirmed_txids = confirmables
+					.iter()
+					.flat_map(|c| c.get_relevant_txids())
+					.filter(|txid| {
+						client
+							.get_tx_status(txid)
+							.ok()
+							.unwrap_or(None)
+							.map_or(true, |status| !status.confirmed)
+					})
+					.collect::<Vec<Txid>>();
+
+				// Mark all relevant unconfirmed transactions as unconfirmed.
+				for txid in &unconfirmed_txids {
+					for c in &confirmables {
+						c.transaction_unconfirmed(txid);
+					}
+				}
+			}
+		}
+
+		// TODO: check whether new outputs have been registered by now and process them
+		Ok(())
+	}
+
+	pub(crate) fn create_funding_transaction(
+		&self, output_script: &Script, value_sats: u64, confirmation_target: ConfirmationTarget,
+	) -> Result<Transaction, Error> {
+		let num_blocks = num_blocks_from_conf_target(confirmation_target);
+		let fee_rate = self.blockchain.estimate_fee(num_blocks)?;
+
+		let locked_wallet = self.wallet.lock().unwrap();
+		let mut tx_builder = locked_wallet.build_tx();
+
+		tx_builder.add_recipient(output_script.clone(), value_sats).fee_rate(fee_rate).enable_rbf();
+
+		let (mut psbt, _) = tx_builder.finish()?;
+		log_trace!(self.logger, "Created funding PSBT: {:?}", psbt);
+
+		// We double-check that no inputs try to spend non-witness outputs. As we use a SegWit
+		// wallet descriptor this technically shouldn't ever happen, but better safe than sorry.
+		for input in &psbt.inputs {
+			if input.witness_utxo.is_none() {
+				log_error!(self.logger, "Tried to spend a non-witness funding output. This must not ever happen. Panicking!");
+				panic!("Tried to spend a non-witness funding output. This must not ever happen.");
+			}
+		}
+
+		if !locked_wallet.sign(&mut psbt, SignOptions::default())? {
+			return Err(Error::FundingTxCreationFailed);
+		}
+
+		Ok(psbt.extract_tx())
+	}
+
+	pub(crate) fn get_new_address(&self) -> Result<bitcoin::Address, Error> {
+		let address_info = self.wallet.lock().unwrap().get_address(AddressIndex::New)?;
+		Ok(address_info.address)
+	}
+}
+
+impl<D> FeeEstimator for LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 {
+		let num_blocks = num_blocks_from_conf_target(confirmation_target);
+		let fallback_fee = fallback_fee_from_conf_target(confirmation_target);
+		self.blockchain
+			.estimate_fee(num_blocks)
+			.map_or(fallback_fee, |fee_rate| (fee_rate.fee_wu(1000) as u32).max(MIN_FEERATE)) as u32
+	}
+}
+
+impl<D> BroadcasterInterface for LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn broadcast_transaction(&self, tx: &Transaction) {
+		match self.blockchain.broadcast(tx) {
+			Ok(_) => {}
+			Err(err) => log_error!(self.logger, "Failed to broadcast transaction: {}", err),
```

Comment:
Well, this is what's happening? I don't recall where your and @TheBlueMatt's discussion stopped: should we panic in this case, or are we fine just logging it an waiting for the eventual re-broadcast?

Reply:
Afaik, from LDK perspective this tx has been successfully broadcasted, so it will not be re-broadcasted again.
Idk if LDK will keep on waiting for it to confirm but since it was never broadcasted, it will never confirm.

For now we might want to panic?